plan/pattern.go

```diff
@@ -20,8 +20,18 @@ type Pattern interface {
 //  /   \
 // A     B
 func Pat(kind ProcedureKind, predecessors ...Pattern) Pattern {
-	return &OneKindPattern{
-		kind:         kind,
+	return &UnionKindPattern{
+		kinds:        []ProcedureKind{kind},
+		predecessors: predecessors,
+	}
+}
+
+// Union pat matches any plan node from a given set of ProcedureKind and whose
+// predecessors match the given predecessor patterns. This is idencal to Pat,
```

Comment:
Typo: idencal. Also, comments should start with the exact name of the function so this should be `UnionPat` instead  of `Union pat`.